http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -1247,83 +1261,36 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
         .transaction
 
       // Poll for namespace deletion
-      op = F.blocking(
-        wsmApi.getDeleteAzureKubernetesNamespaceResult(workspaceId.value, jobId.toString)
-      )
-      result <- streamFUntilDone(
-        op,
-        config.appMonitorConfig.deleteApp.maxAttempts,
-        config.appMonitorConfig.deleteApp.interval
-      ).compile.lastOrError
-
-      _ <- logger.info(ctx.loggingCtx)(s"WSM delete namespace job result: ${result}")
-
-      _ <- F.raiseWhen(result.getJobReport.getStatus != JobReport.StatusEnum.SUCCEEDED)(
-        AppDeletionException(
-          s"WSM namespace deletion failed for app ${app.appName.value}. WSM response: $result"
-        )
-      )
-
-      // Update record in APP_CONTROLLED_RESOURCE table
-      _ <- appControlledResourceQuery.delete(wsmResource.resourceId).transaction
-    } yield ()
-
-  private[util] def deleteAzureDatabaseResource(workspaceId: WorkspaceId,
-                                                app: App,
-                                                wsmResource: AppControlledResourceRecord
-  )(implicit
-    ev: Ask[F, AppContext]
-  ): F[Unit] =
-    for {
-      ctx <- ev.ask
-
-      // Build WSM client
-      wsmApi <- buildWsmControlledResourceApiClient
-
-      // Build delete database request
-      jobId = UUID.randomUUID()
-      deleteDatabaseRequest = new DeleteControlledAzureResourceRequest().jobControl(
-        new JobControl().id(jobId.toString)
-      )
-
-      _ <- logger.info(ctx.loggingCtx)(
-        s"WSM delete database request for app ${app.appName.value}: $deleteDatabaseRequest"
-      )
-
-      // Execute WSM call
-      result <- F.blocking(
-        wsmApi.deleteAzureDatabaseAsync(deleteDatabaseRequest, workspaceId.value, wsmResource.resourceId.value)
-      )
-
-      _ <- logger.info(ctx.loggingCtx)(s"WSM delete database response for app ${app.appName.value}: $result")
-
-      // Update record in APP_CONTROLLED_RESOURCE table
-      _ <- appControlledResourceQuery
-        .updateStatus(
-          wsmResource.resourceId,
-          AppControlledResourceStatus.Deleting
-        )
-        .transaction
+      op = wsmResource.resourceType match {
+        case WsmResourceType.AzureKubernetesNamespace =>
+          F.blocking(
+            wsmApi.getDeleteAzureKubernetesNamespaceResult(workspaceId.value, jobId.toString)
+          )
+        case WsmResourceType.AzureDatabase =>
+          F.blocking(
+            wsmApi.getDeleteAzureDatabaseResult(workspaceId.value, jobId.toString)
+          )
+        case _ =>
+          F.raiseError[DeleteControlledAzureResourceResult](
+            AppDeletionException(s"Unexpected WSM resource type for deletion polling ${wsmResource.resourceType}")
+          )
+      }
```

Comment:
in theory, we will never get here because of the earlier default condition.
question ... 
does it make sense to collapse the case conditions and do the two operations side-by-side?
proposing order of operations being ...
- update our database state to DELETING
- then issue commands (make request, and poll) in single case context
- followed by cleanup code below